src/main/java/io/quarkus/gizmo/ClassSignatureBuilderImpl.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.gizmo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.quarkus.gizmo.SignatureBuilder.ClassSignatureBuilder;
+import io.quarkus.gizmo.Type.ClassType;
+import io.quarkus.gizmo.Type.ParameterizedType;
+import io.quarkus.gizmo.Type.TypeVariable;
+
+class ClassSignatureBuilderImpl implements ClassSignatureBuilder {
+    private List<TypeVariable> typeParameters = new ArrayList<>();
+    private Type superClass = ClassType.OBJECT;
+    private List<Type> superInterfaces = new ArrayList<>();
+
+    @Override
+    public String build() {
+        StringBuilder signature = new StringBuilder();
+
+        // type params
+        if (!typeParameters.isEmpty()) {
+            signature.append('<');
+            for (TypeVariable typeParameter : typeParameters) {
+                typeParameter.appendTypeParameterToSignature(signature);
+            }
+            signature.append('>');
+        }
+
+        // superclass
+        superClass.appendToSignature(signature);
+
+        // interfaces
+        if (!superInterfaces.isEmpty()) {
+            for (Type superInterface : superInterfaces) {
+                superInterface.appendToSignature(signature);
+            }
+        }
+        return signature.toString();
+    }
+
+    @Override
+    public ClassSignatureBuilder addTypeParameter(TypeVariable typeParameter) {
+        typeParameters.add(typeParameter);
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder setSuperClass(ClassType superClass) {
```

Comment:
Given `Class.getSuperclass`, I would rather keep the same case for `Superclass`. Or is it written this way in other parts of Gizmo?

Reply:
Ah I see that `SuperClass` is used in other parts of Gizmo. That's a bit unfortunate but I suppose it's too late to change it.

Reply:
I don't understand. Where is it used? And what exactly would you like to keep?

Reply:
My point was that in the JDK APIs, it's `Superclass`, not `SuperClass`. But Gizmo is using `SuperClass` in other places so we should probably keep it as is.
Is it clearer?

Reply:
:+1: I see..